server/CommandService/Services/SourceService.cs

```diff
@@ -0,0 +1,35 @@
+using System;
+using System.IO;
+using System.Threading.Tasks;
+using Commands;
+using Microsoft.AspNetCore.Http;
+using RawRabbit;
+
+namespace CommandService.Services
+{
+    public class SourceService : ISourceService
+    {
+        private readonly IBusClient _client;
+
+        public SourceService(IBusClient client)
+        {
+            _client = client;
+        }
+
+        public async Task<Guid> CreateSource(IFormFile file, Guid userId)
+        {
+            var taskId = Guid.NewGuid();
+            var fileModel = await ConvertToFileModel(file);
+            var command = new AddDocumentToSourceStoreCommand(taskId, userId, fileModel);
+            await _client.PublishAsync(command);
+            return taskId;
+        }
+
+        private async Task<FileModel> ConvertToFileModel(IFormFile file)
```

Comment:
this could be a constructor for FileModel (but couldn't be async then) or an extenstion not to duplicate this fragment

Reply:
Adding constructor will force Commands to be dependent from Microsoft.AspNetCore.Http, so I think using extension will be better here